clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -2479,23 +2487,29 @@ public void testCurrentLag(GroupProtocol groupProtocol) {
         consumer.seek(tp0, 50L);
         consumer.poll(Duration.ofMillis(0));
         // requests: list-offset, fetch
-        assertEquals(2, client.inFlightRequestCount());
```

Comment:
similar to the reasoning that led to this fix, there is a check above for:
> assertEquals(0, client.inFlightRequestCount())
I would say that may be flaky for the new consumer, right? 

We cannot ensure that no requests will be generated just because the last api we called shouldn't generate one. Given that we called poll before (and there is a background thread running) I expect there could be requests generated for fetching and fetching offsets maybe. The point of that assert is to ensure there was no ListOffsets request generated, so I would say that we should look for that exact request type, and assert there is none. Makes sense?

Reply:
Hi @lianetm, thanks for catching this. I tried to add `Thread.sleep(1000)` before `assertEquals(0, client.inFlightRequestCount())`. It looks like both `ListOffsetsRequest` and `OffsetFetchRequest` will be sent after getting `FindCoordinatorResponse`. I remove the assertion now. Do you think that we should keep this assertion for CLASSIC group protocol only? Thank you.

Reply:
Hi @lianetm, I added the assertion back and comments. I run 50 rounds on my laptop and there is no error. Let's latest CI result. Thank you.

Reply:
Hey @FrankYang0529, this testCurrentLag still seems flaky even after the latest changes, I filed https://issues.apache.org/jira/browse/KAFKA-17560 with what I see and where I imagine the flakiness may be, but it needs more thinking probably. Here's a suggestion to make progress: 

- we could leave `testCurrrentLag` disabled for the new consumer on this PR (just as it was before)
- we unblock the other 2 tests, that seems to pass consistently after your changes
- we address `testCurrentLag` in a separate PR, with that jira I created (including the changes you had here, just that I think it needs more)

What do you think?

Reply:
Hi @lianetm, thanks for filing the Jira. I would like to give "eventually updated" a try in this PR. If it still can't make the test stable, we can put it in next Jira. What do you think? Thank you.

Reply:
With latest update, I run `I=0;  while ./gradlew cleanTest :clients:test --tests KafkaConsumerTest --rerun --fail-fast; do  (( I=$I+1 )); echo "Completed run: $I"; sleep 1; done` 100 times without error.

<img width="1311" alt="Screenshot 2024-09-17 at 9 26 24 AM" src="https://github.com/user-attachments/assets/01254142-3e0a-4d7f-b604-8639d24dbb91">

Reply:
Hi @lianetm, do you think I need to rebase code, so we can have more CI running result? Or, it's good to keep current state? Thank you.